app/models/verified_principal.rb

```diff
@@ -0,0 +1,35 @@
+class VerifiedPrincipal
+  attr_accessor :form, :firm_name
+
+  def initialize(form_data, firm_name)
+    @form = NewPrincipalForm.new(form_data)
+    @firm_name = firm_name
+  end
+
+  def register!
+    create_new_principal
+    @user.principal.firm = create_associate_firm
```

Comment:
If the principal has the primary key of fca number and firm the foreign key of fca number, how relevant is the assign of `@user.principal.firm= `? Can we not just call the `create_associate_firm`?